### ISSUE TITLE
[BUG]  incompatible with pandas 2.2.1

### ISSUE THREAD

**Issue by qiuwei** [pwwang/datar#208]
### datar version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the **latest version** of datar and its backends.

### Issue Description

The following code throws with pandas 2.2.1, but runs without any issue with pandas 2.0.3
Minimal working example to reproduce
sample.csv
[sample.csv](https://github.com/pwwang/datar/files/14468431/sample.csv)

```
from datar.all import *

import pandas as pd

df = pd.read_csv("sample.csv")

print(df >> group_by(f.session))

```

which throws
```
traceback (most recent call last):
  File "/home/johndoe/celere/scripts/test_datar.py", line 8, in <module>
    print(df >> group_by(f.session))
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 1203, in __repr__
    return self.to_string(**repr_params)
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 1383, in to_string
    return fmt.DataFrameRenderer(formatter).to_string(
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/io/formats/format.py", line 962, in to_string
    string = string_formatter.to_string()
  File "/home/johndoe/.local/lib/python3.10/site-packages/pdtypes/formatters.py", line 337, in to_string
    text = super().to_string()
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/io/formats/string.py", line 29, in to_string
    text = self._get_string_representation()
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/io/formats/string.py", line 53, in _get_string_representation
    return self._fit_strcols_to_terminal_width(strcols)
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/io/formats/string.py", line 184, in _fit_strcols_to_terminal_width
    self.fmt.truncate()
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/io/formats/format.py", line 658, in truncate
    self._truncate_vertically()
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/io/formats/format.py", line 698, in _truncate_vertically
    self.tr_frame = self.tr_frame.iloc[_slice]
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/core/indexing.py", line 1191, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/core/indexing.py", line 1743, in _getitem_axis
    return self._get_list_axis(key, axis=axis)
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/core/indexing.py", line 1714, in _get_list_axis
    return self.obj._take_with_is_copy(key, axis=axis)
  File "/home/johndoe/.local/lib/python3.10/site-packages/pandas/core/generic.py", line 4150, in _take_with_is_copy
    result = self.take(indices=indices, axis=axis)
  File "/home/johndoe/.local/lib/python3.10/site-packages/datar_pandas/tibble.py", line 370, in take
    return result.regroup(inplace=False)
  File "/home/johndoe/.local/lib/python3.10/site-packages/datar_pandas/tibble.py", line 326, in regroup
    grouped = self._datar["grouped"]
KeyError: 'grouped'
```

However, if I revert pandas to 2.0.3, the above code runs without any issue.

The strange behavior is also related to the data, if I remove the last line of the sample data, then the code runs fine with pandas 2.2.1. 

### Expected Behavior

The above code runs without throwing. 
Or datar specifies the compatible pandas' version in the package metadata.

### Installed Versions

<details>
python      : 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
datar       : 0.15.5
simplug     : 0.3.2
executing   : 2.0.1
pipda       : 0.13.1
datar-numpy : 0.3.4
numpy       : 1.26.4
datar-pandas: 0.5.4
pandas      : 2.2.1
</details>

**Comment by pwwang** [comment]
I can reproduce the exact behavior. Digging into it.

**Comment by pwwang** [comment]
It's fixed by [`datar-pandas` v0.5.5](https://github.com/pwwang/datar-pandas/releases/tag/0.5.5).

Try `pip install -U datar-pandas` or `pip install -U datar[pandas]`.

It's because the dataframe is truncated with 61 rows when printed. But a 60-row dataframe is not truncated. When truncating the dataframe, the metadata is lost. 

Note that the truncating is not necessarily related to number of rows. It's also related to the terminal size.
